organization_token.go

```diff
@@ -53,6 +67,35 @@ type OrganizationTokenCreateOptions struct {
 	// Optional: The token's expiration date.
 	// This feature is available in TFE release v202305-1 and later
 	ExpiredAt *time.Time `jsonapi:"attr,expired-at,iso8601,omitempty"`
+	// Optional: What type of token to create
+	// This option is only applicable to HCP Terraform and is ignored by TFE.
+	TokenType *TokenType
+}
+
+// OrganizationTokenReadOptions contains the options for reading an organization token.
+type OrganizationTokenReadOptions struct {
+	// Optional: What type of token to read
+	// This option is only applicable to HCP Terraform and is ignored by TFE.
+	TokenType *TokenType
```

Comment:
If you annotate these with `url:"token,omitempty"` the underlying library will automatically encode the token type into the query string for each query (if it is not empty). This may be a better choice because it would make extending the parameters easier in the future. I don't think what you have is blocking, though.

Reply:
So, I did look at that and unfortunately the annotations are ignored for DELETE and POST calls (tfe.go:273+ https://github.com/hashicorp/go-tfe/blob/fd766c164d112e5cca222d8412c58d83716ab8dc/tfe.go#L287)

Which is why I had to inject the query string (which I did not like doing :-( )

Reply:
Would you prefer me to update that helper to honor the url annotations ?

It's basically moving that logic out of the "GET" case to before/after the switch

Reply:
Rebased on the new request helpers